client/http.go

```diff
@@ -0,0 +1,88 @@
+package client
+
+import (
+	"bytes"
+	"context"
+	"encoding/json"
+	"fmt"
+	"net/http"
+	"time"
+
+	"github.com/drand/drand/beacon"
+	"github.com/drand/drand/key"
+)
+
+// NewHTTPClient creates a new client pointing to an HTTP endpoint
+func NewHTTPClient(url string, groupHash []byte, client *http.Client) (Client, error) {
```

Comment:
Would it be easier to use if `client` was optional. i.e. it would just use `http.Get` unless a `http.Client` is passed?

Reply:
a null client can be passed, which will be equivalent to using the global http requester